pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceBaseArguments.java

```diff
@@ -19,53 +19,44 @@
 package org.apache.pulsar.testclient;
 
 import static org.apache.commons.lang3.StringUtils.isBlank;
-import static org.apache.pulsar.testclient.PerfClientUtils.exit;
-import java.io.File;
-import java.io.FileInputStream;
-import java.util.Properties;
-import lombok.SneakyThrows;
-import org.apache.commons.lang3.StringUtils;
 import org.apache.pulsar.cli.converters.picocli.ByteUnitToLongConverter;
 import org.apache.pulsar.client.api.ProxyProtocol;
-import picocli.CommandLine;
 import picocli.CommandLine.Option;
-import picocli.CommandLine.ParameterException;
 
 /**
  * PerformanceBaseArguments contains common CLI arguments and parsing logic available to all sub-commands.
  * Sub-commands should create Argument subclasses and override the `validate` method as necessary.
  */
-public abstract class PerformanceBaseArguments {
+public abstract class PerformanceBaseArguments extends CmdBase{
```

Comment:
Please move `validate()` to `CmdBase`, add an override in the `PerformanceBaseArguments`, move the `parseCLI` logic to that.

Otherwise, this code will look a bit confusing.

Reply:
Good suggestion. Done.